crates/uv/src/commands/tool/run.rs

```diff
@@ -448,66 +511,80 @@ async fn get_or_create_environment(
-            source: RequirementSource::Registry {
-                specifier: VersionSpecifiers::from(VersionSpecifier::equals_version(
-                    version.clone(),
-                )),
-                index: None,
-                conflict: None,
+            // Ex) `ruff@0.6.0`
+            Target::Version(name, version) | Target::FromVersion(_, name, version) => Requirement {
+                name: PackageName::from_str(name)?,
+                extras: vec![],
+                groups: vec![],
+                marker: MarkerTree::default(),
+                source: RequirementSource::Registry {
+                    specifier: VersionSpecifiers::from(VersionSpecifier::equals_version(
+                        version.clone(),
+                    )),
+                    index: None,
+                    conflict: None,
+                },
+                origin: None,
             },
-            origin: None,
-        },
-        // Ex) `ruff@latest`
-        Target::Latest(name) | Target::FromLatest(_, name) => Requirement {
-            name: PackageName::from_str(name)?,
-            extras: vec![],
-            groups: vec![],
-            marker: MarkerTree::default(),
-            source: RequirementSource::Registry {
-                specifier: VersionSpecifiers::empty(),
-                index: None,
-                conflict: None,
+            // Ex) `ruff@latest`
+            Target::Latest(name) | Target::FromLatest(_, name) => Requirement {
+                name: PackageName::from_str(name)?,
+                extras: vec![],
+                groups: vec![],
+                marker: MarkerTree::default(),
+                source: RequirementSource::Registry {
+                    specifier: VersionSpecifiers::empty(),
+                    index: None,
+                    conflict: None,
+                },
+                origin: None,
             },
-            origin: None,
-        },
-        // Ex) `ruff>=0.6.0`
-        Target::From(_, from) => resolve_names(
-            vec![RequirementsSpecification::parse_package(from)?],
-            &interpreter,
-            settings,
-            &state,
-            connectivity,
-            concurrency,
-            native_tls,
-            allow_insecure_host,
-            cache,
-            printer,
-            preview,
-        )
-        .await?
-        .pop()
-        .unwrap(),
+            // Ex) `ruff>=0.6.0`
+            Target::From(_, from) => {
+                let spec = RequirementsSpecification::parse_package(from)?;
+                if let UnresolvedRequirement::Named(requirement) = &spec.requirement {
+                    if requirement.name.as_str() == "python" {
+                        return Err(anyhow::anyhow!(
+                            "Using `--from python<specifier>` is not supported. Use `python@<version>` instead."
+                        )
+                        .into());
+                    }
+                }
```

Comment:
This is new, to prevent confusing errors about not finding `python` in the registry.